data-prepper-plugins/opensearch-source/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/ElasticsearchAccessorTest.java

```diff
@@ -159,6 +162,27 @@ void create_pit_with_exception_for_pit_limit_throws_SearchContextLimitException(
         assertThrows(SearchContextLimitException.class, () -> createObjectUnderTest().createPit(createPointInTimeRequest));
     }
 
+    @Test
+    void create_pit_with_exception_for_index_not_found_throws_IndexNotFoundException() throws IOException {
+        final String indexName = UUID.randomUUID().toString();
+        final String keepAlive = UUID.randomUUID().toString();
+
+        final CreatePointInTimeRequest createPointInTimeRequest = mock(CreatePointInTimeRequest.class);
+        when(createPointInTimeRequest.getIndex()).thenReturn(indexName);
+        when(createPointInTimeRequest.getKeepAlive()).thenReturn(keepAlive);
+
+        final ElasticsearchException exception = mock(ElasticsearchException.class);
+        final ErrorResponse errorResponse = mock(ErrorResponse.class);
+        final ErrorCause errorCause = mock(ErrorCause.class);
+        when(errorCause.type()).thenReturn(INDEX_NOT_FOUND_EXCEPTION);
+        when(errorResponse.error()).thenReturn(errorCause);
+        when(exception.response()).thenReturn(errorResponse);
+
+        when(elasticSearchClient.openPointInTime(any(OpenPointInTimeRequest.class))).thenThrow(exception);
+
+        assertThrows(IndexNotFoundException.class, () -> createObjectUnderTest().createPit(createPointInTimeRequest));
```

Comment:
I recommend making the `assertThrows` runnable take the smallest bit of coded needed. So for there, move `createObjectUnderTest()` out to the new line. This way, you know exactly what is throwing.

```
ElasticsearchAccessor objectUnderTest = createObjectUnderTest();
 assertThrows(IndexNotFoundException.class, () -> objectUnderTest.createPit(createPointInTimeRequest));
```

There are a few other places where you can do something similar.